src/classrooms/classrooms.service.ts

```diff
@@ -93,4 +93,17 @@ export class ClassroomsService {
     await this.em.removeAndFlush(classroom);
     return true;
   }
+
+  async updateClassroom(
+    id: number,
+    updateClassroomDto: UpdateClassroomDto,
+    userId: number,
+  ): Promise<Classroom | null> {
+    const classroom = await this.getClassroomById(id, userId);
```

Comment:
Maybe consider renaming the method getClassroomById as it is doing more than simply getting classroom by id.